x/ecocredit/client/tx.go

```diff
@@ -49,17 +56,152 @@ func txCreateClass() *cobra.Command {
+			msg := ecocredit.MsgCreateClassRequest{
 				Designer: args[0], Issuers: issuers, Metadata: b,
-			})
-			return mkTx(ctx, cmd, err)
+			}
+			_, err = c.client.CreateClass(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txCreateBatch() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create_batch [issuer] [class_id] [metadata] [issuance]",
+		Short: "Issues a new credit batch",
+		Long: `Issues a new credit batch.
+
+Parameters:
+  issuer:    issuer address
+  class_id:  credit class
+  metadata:  base64 encoded issuance metadata
+  issuance:  JSON encode issuance list,
+             eg: '[{"recipient": "a1", "tradableUnits": "10", "retiredUnits": "2"}]'
+             Note: "tradableUnits" and "retiredUnits" default to 0.`,
+		Args: cobra.ExactArgs(4),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			b, err := base64.StdEncoding.DecodeString(args[2])
+			if err != nil {
+				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "metadata is malformed, proper base64 string is required")
+			}
+			var issuance = []*ecocredit.MsgCreateBatchRequest_BatchIssuance{}
+			if err = json.Unmarshal([]byte(args[3]), &issuance); err != nil {
+				return err
+			}
+
+			msg := ecocredit.MsgCreateBatchRequest{
+				Issuer: args[0], ClassId: args[1], Metadata: b, Issuance: issuance,
+			}
+			c, err := newMsgSrvClient(cmd)
+			if err != nil {
+				return err
+			}
+			_, err = c.client.CreateBatch(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txSend() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "send [recipient] [credits]",
+		Short: "Sends credits from the transaction author (--from) to the recipient",
+		Long: `Sends credits from the transaction author (--from) to the recipient.
+
+Parameters:
+  recipient: recipient address
+  credits:   JSON encoded credit list
+             eg: '[{"batchDenom": "100/2", "tradableUnits": "5", "retiredUnits": "0"}]'`,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var credits = []*ecocredit.MsgSendRequest_SendUnits{}
+			if err := json.Unmarshal([]byte(args[1]), &credits); err != nil {
+				return err
+			}
+			c, err := newMsgSrvClient(cmd)
+			if err != nil {
+				return err
+			}
+			msg := ecocredit.MsgSendRequest{
+				Sender:    c.Cctx.GetFromAddress().String(),
+				Recipient: args[0], Credits: credits,
+			}
+			_, err = c.client.Send(cmd.Context(), &msg)
+			return c.send(err)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func txRetire() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "retire [credits]",
+		Short: "Retires a specified amounts of credits from the account of the transaction author (--from)",
+		Long: `Retires a specified amounts of credits from the account of the transaction author (--from)
+
+Parameters:
+  recipient: recipient address
```

Comment:
```suggestion
```
this is unneeded right?

Reply:
Yes, I the previous implementation I had this parameter. Thanks for catching.